internal/api/user_devices.go

```diff
@@ -267,6 +267,37 @@ func (s *userDeviceService) RegisterUserDeviceFromVIN(ctx context.Context, req *
 	return &pb.RegisterUserDeviceFromVINResponse{Created: true}, err
 }
 
+func (s *userDeviceService) UpdateDeviceIntegrationStatus(ctx context.Context, req *pb.UpdateDeviceIntegrationStatusRequest) (*pb.UserDevice, error) {
+	dbDevice, err := models.UserDevices(
+		models.UserDeviceWhere.ID.EQ(req.UserDeviceId),
+		qm.Load(
+			models.UserDeviceRels.UserDeviceAPIIntegrations,
+			models.UserDeviceAPIIntegrationWhere.IntegrationID.EQ(req.IntegrationId),
+		),
+	).One(ctx, s.dbs().Reader)
+
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return nil, status.Error(codes.NotFound, "No UserDeviceAPIIntegrations with that ID found.")
+		}
+
+		s.logger.Err(err).Str("IntegrationId", req.IntegrationId).Msg("Database failure retrieving UserDeviceAPIIntegrations.")
+		return nil, status.Error(codes.Internal, "Internal error.")
+	}
+
+	if len(dbDevice.R.UserDeviceAPIIntegrations) == 0 {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("can't find API integration for device %s and integration %s", req.UserDeviceId, req.IntegrationId))
+	}
+
+	apiIntegration := dbDevice.R.UserDeviceAPIIntegrations[0]
```

Comment:
there could be more than one integration, ie. for smartcar or autopi. 
Add the `integrationId` to the `req` so you can select the right `dbDevice.R.UserDeviceAPIIntegrations`